### ISSUE TITLE
Support Training model with Keras Embedding layer

### ISSUE THREAD

**Issue by mhaoli** [sql-machine-learning/elasticdl#1004]
In our implementation now, the gradients are supposed to be in the form of `tf.Tensor`. However the gradients of keras Embedding layer are `tf.IndexedSlices`. 

[`tf.IndexedSlices`](https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/IndexedSlices) is used for sparse update.
